connectors/src/connectors/snowflake/temporal/activities.ts

```diff
@@ -1,7 +1,157 @@
-export async function syncSnowflakeConnection(connectorId: number) {
-  // TODO(SNOWFLAKE)
-  // do SHOW TABLES on the Snowflake connection
-  // look at DBs / Schemas / Tables with explicit read permissions
-  // garbage collect / add missing resources as needed
-  void connectorId;
+import type { ModelId } from "@dust-tt/types";
+import { isLeft } from "fp-ts/lib/Either";
+import * as t from "io-ts";
+import * as reporter from "io-ts-reporters";
+
+import { fetchTables } from "@connectors/connectors/snowflake/lib/snowflake_api";
+import { getConnectorAndCredentials } from "@connectors/connectors/snowflake/lib/utils";
+import { dataSourceConfigFromConnector } from "@connectors/lib/api/data_source_config";
+import { deleteTable, upsertTable } from "@connectors/lib/data_sources";
+import {
+  RemoteDatabaseModel,
+  RemoteSchemaModel,
+  RemoteTableModel,
+} from "@connectors/lib/models/remote_databases";
+import { syncSucceeded } from "@connectors/lib/sync_status";
+import logger from "@connectors/logger/logger";
+
+const snowflakeTableCodec = t.type({
+  name: t.string,
+  database_name: t.string,
+  schema_name: t.string,
+});
+
+export async function syncSnowflakeConnection(connectorId: ModelId) {
+  const getConnectorAndCredentialsRes = await getConnectorAndCredentials({
+    connectorId,
+    logger,
+  });
+  if (getConnectorAndCredentialsRes.isErr()) {
+    throw getConnectorAndCredentialsRes.error;
+  }
+
+  const { credentials, connector } = getConnectorAndCredentialsRes.value;
+
+  const tablesRes = await fetchTables({ credentials });
+  if (tablesRes.isErr()) {
+    throw tablesRes.error;
+  }
+
+  const tablesValidation = t.array(snowflakeTableCodec).decode(tablesRes.value);
+  if (isLeft(tablesValidation)) {
+    const pathError = reporter.formatValidationErrors(tablesValidation.left);
+    throw new Error(`Invalid tables response: ${pathError}`);
+  }
+  const [allDatabases, allSchemas, allTables] = await Promise.all([
```

Comment:
nit: I would prefix the things coming from snowflake vs the things coming from our DB to make the code more readable.